src/types.tsx

```diff
@@ -0,0 +1,252 @@
+  /** Set Formik state, careful! */
+  setFormikState<K extends keyof FormikState<Values>>(
+    state: Pick<FormikState<Values>, K>,
+    callback?: () => any
+  ): void;
+}
+
+/**
+ * Formik form event handlers
+ */
+export interface FormikHandlers {
+  /** Form submit handler */
+  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
+  /** Reset form event handler  */
+  handleReset: () => void;
+  /** Classic React blur handler, keyed by input name */
+
+  handleBlur(e: any): void;
+  /** Preact-like linkState. Will return a handleBlur function. */
+  handleBlur(field: string): ((e: any) => void);
+  /** Classic React change handler, keyed by input name */
+  handleChange(e: React.ChangeEvent<any>): void;
+  /** Preact-like linkState. Will return a handleChange function.  */
+  handleChange(field: string): ((e: React.ChangeEvent<any>) => void);
+}
+
+/**
+ * Base formik configuration/props shared between the HoC and Component.
+ */
+export interface FormikSharedConfig {
+  /** Tells Formik to validate the form on each input's onChange event */
+  validateOnChange?: boolean;
+  /** Tells Formik to validate the form on each input's onBlur event */
+  validateOnBlur?: boolean;
+  /** Tell Formik if initial form values are valid or not on first render */
+  isInitialValid?: boolean | ((props: object) => boolean | undefined);
+  /** Should Formik reset the form when new initialValues change */
+  enableReinitialize?: boolean;
+}
+
+/**
+ * <Formik /> props
+ */
+export interface FormikConfig<Values> extends FormikSharedConfig {
+  /**
+   * Initial values of the form
+   */
+  initialValues: Values;
+
+  /**
+   * Reset handler
+   */
+  onReset?: (values: Values, formikActions: FormikActions<Values>) => void;
+
+  /**
+   * Submission handler
+   */
+  onSubmit: (values: Values, formikActions: FormikActions<Values>) => void;
+
+  /**
+   * Form component to render
+   */
+  component?: React.ComponentType<FormikProps<Values>> | React.ReactNode;
+
+  /**
+   * Render prop (works like React router's <Route render={props =>} />)
+   */
+  render?: ((props: FormikProps<Values>) => React.ReactNode);
+
+  /**
+   * A Yup Schema or a function that returns a Yup schema
+   */
+  validationSchema?: any | (() => any);
+
+  /**
+   * Validation function. Must return an error object or promise that
+   * throws an error object where that object keys map to corresponding value.
+   */
+  validate?: ((
+    values: Values
+  ) => void | object | Promise<FormikErrors<Values>>);
+
+  /**
+   * React children or child render callback
+   */
+  children?:
+    | ((props: FormikProps<Values>) => React.ReactNode)
+    | React.ReactNode;
+}
+
+/**
+ * State, handlers, and helpers made available to form component or render prop
+ * of <Formik/>.
+ */
+export type FormikProps<Values> = FormikSharedConfig &
+  FormikState<Values> &
+  FormikActions<Values> &
+  FormikHandlers &
+  FormikComputedProps<Values> & {
+    validate?: ((
+      values: Values
```

Comment:
So in the past, we did not pass thru props or expose Formik's guts to general public. Formik props and context had slight differences. I'm not sure this is a good idea.

Reply:
I wonder if this should just be: 

```tsx
export type FormikContext<Values> = FormikProps<Values> &
  Pick<FormikConfig<Values>, 'validate' | 'validationSchema'> & {
    registerField(name: string, resetFn: (nextValues?: any) => void): void;
    unregisterField(name: string): void;
  };
```

Reply:
I guess we are using the word FormikContext to be the name of teh object returned by `create-react-context`. Technically we have never given this a name, it's just been typed as `any` for the most part. The only weird gotcha is that FastField needs to know about `<Formik>`'s props `validate` and `validationSchema` with current implementation.